src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -30,6 +30,20 @@ internal sealed partial class PkcsPalOpenSsl : PkcsPal
                 throw new PlatformNotSupportedException(SR.Cryptography_Cms_AddOriginatorCertsPlatformNotSupported);
             }
 
+            if (contentEncryptionAlgorithm.KeyLength != 0 && contentEncryptionAlgorithm.KeyLength != KeyLengths.DefaultKeyLengthForRc2AndRc4)
+            {
+                switch (contentEncryptionAlgorithm.Oid.Value)
+                {
+                    case Oids.Rc2:
+                        throw new PlatformNotSupportedException(SR.Cryptography_Cms_Rc2VariableKeyLengthPlatformNotSupported);
+                    case Oids.Rc4:
+                        throw new PlatformNotSupportedException(SR.Cryptography_Cms_Rc4VariableKeyLengthPlatformNotSupported);
+                    default:
+                        // RC2 and RC4 are the only ciphers for which we support a variable key length that we need to worry about.
```

Comment:
So requesting Oids.Aes128 and specifying a keysize of 3 isn't an error at all? Is that also true for Windows? And do we have a test for that?